### ISSUE TITLE
Matching by sector, returns results with mismatched sectors

### ISSUE THREAD

**Issue by jdhoffa** [2DegreesInvesting/r2dii.match#111]
When running `match_name(...,by_sector=TRUE)`, results with non-matching sectors are still output:

``` r
library(r2dii.match)
library(r2dii.dataraw)
#> Loading required package: r2dii.utils
suppressPackageStartupMessages(
  library(tidyverse)
)

mismatched_sector <- match_name(loanbook_demo, ald_demo, by_sector = TRUE) %>% 
  filter(sector != sector_ald)
interesting_columns <- c("id", "level", "name", "name_ald", "sector", "sector_ald")

head(select(mismatched_sector, interesting_columns))
#> # A tibble: 6 x 6
#>   id    level       name                name_ald             sector   sector_ald
#>   <chr> <chr>       <chr>               <chr>                <chr>    <chr>     
#> 1 UP230 ultimate_p… San Yang Industry … san yang industry c… automot… shipping  
#> 2 DL172 direct_loa… Shanghai Automotiv… shanghai automotive… automot… shipping  
#> 3 DL193 direct_loa… Subaru Corp         subaru corp          automot… power     
#> 4 DL196 direct_loa… Suzuki Motor Corp   suzuki motor corp    automot… shipping  
#> 5 DL196 direct_loa… Suzuki Motor Corp   suzuki motor corp    automot… power     
#> 6 DL200 direct_loa… Tata Group          tata group           steel    automotive
```

<sup>Created on 2020-02-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by jdhoffa** [comment]
Ok, this happens because `by_sector = TRUE` is utilized in `score_alias_similarity`, meaning that the name is only compared against names in the ALD having the same sector. 

However it is possible that there are two identical names in the ALD with different sectors, thus when joining the data back in to the loanbook, we need to include the by_sector flag as well. 

Fixing now...

**Comment by maurolepore** [comment]
This was closed by #112
